### ISSUE TITLE
conda forge build issues with 0.8.0 release

### ISSUE THREAD

**Issue by cyrush** [LLNL/conduit#893]
linux builds: python 3.8 and 3.9 builds do something different with python path:

```
CMake Error at cmake/thirdparty/SetupPython.cmake:164 (MESSAGE):
  Failed to find main library using
  PYTHON_EXECUTABLE=/home/conda/feedstock_root/build_artifacts/conduit_1640064348711/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_pla/bin/python
Call Stack (most recent call first):
  cmake/Setup3rdParty.cmake:77 (include)
  CMakeLists.txt:83 (include)
```

Python 3.8 build issue on windows only:
```
conduit_python.cpp
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(3184): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(3184): error C2440: 'initializing': cannot convert from 'const char [20]' to 'int'
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(3184): note: There is no context in which this conversion is possible
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8085): error C2027: use of undefined type 'module_state'
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8034): note: see declaration of 'module_state'
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8088): error C2027: use of undefined type 'module_state'
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8034): note: see declaration of 'module_state'
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8098): error C2065: 'PyConduit_DataType_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8103): error C2065: 'PyConduit_Schema_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8108): error C2065: 'PyConduit_Generator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8113): error C2065: 'PyConduit_NodeIterator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8118): error C2065: 'PyConduit_Node_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8123): error C2065: 'PyConduit_Endianness_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8133): error C2065: 'PyConduit_DataType_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8136): error C2065: 'PyConduit_DataType_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8141): error C2065: 'PyConduit_Schema_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8144): error C2065: 'PyConduit_Schema_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8150): error C2065: 'PyConduit_Generator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8153): error C2065: 'PyConduit_Generator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8159): error C2065: 'PyConduit_NodeIterator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8162): error C2065: 'PyConduit_NodeIterator_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8168): error C2065: 'PyConduit_Node_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8171): error C2065: 'PyConduit_Node_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8178): error C2065: 'PyConduit_Endianness_TYPE': undeclared identifier
%SRC_DIR%\src\libs\conduit\python\conduit_python.cpp(8181): error C2065: 'PyConduit_Endianness_TYPE': undeclared identifier
NMAKE : fatal error U1077: 'D:\bld\conduit_1640064548182\_build_env\Library\bin\cmake.exe' : return code '0x2'

```

**Comment by cyrush** [comment]
Python 3.8 windows issue is related to warnings pragma for tp_print field.

**Comment by cyrush** [comment]
python finding issue seems to be related to use of env vars in paths passed during the condaforge build..

While CMake can interpret them, our CMake python inspection fires off python process which will not seem them. We could try to resolve in CMake, but I think the best approach is to resolve them in the shell before they are handed to CMake.

**Comment by cyrush** [comment]
the python finding issue is not an env var thing, in the conda build env python reports the static lib in `LDLIBRARY`, however the static lib is not installed.

To get it installed, you have to request `libpython-static`. 

Trying this, I hit conflicts with with the conda solve.

**Comment by cyrush** [comment]
resolve python find issue with update to conda forge recipe (needed some extra constraints, since static libs pivot exists for python 3.8 >= )

Windows issue should be solved by removing chicken and egg treatment of warnings, but I can test again.

**Comment by cyrush** [comment]
resolved in #910
